packages/app/ui/buttons/CreateOfferBtn.tsx

```diff
@@ -0,0 +1,299 @@
+            setMessage(undefined);
+            setLoading(false);
+
+            const { status } = response as { status: string };
+            if (status === "success") {
+              hideModal();
+              toast.show("You have successfully created an offer", {
+                type: "success",
+              });
+              revalidateOffers();
+            } else {
+              hideModal();
+              reportError(
+                Error(
+                  "Something went wrong during signing and submitting transaction in your wallet."
+                )
+              );
+            }
+          } catch (ex) {
+            hideModal();
+            reportError(ex);
+          }
+        }
+      }
+    };
+
+    const initialValues: CreateOfferForm = {
+      price: "",
+      equityForSale: 1,
+    };
+    return (
+      <>
+        <Modal visible={visible} transparent>
+          <Pressable
+            onPress={() => {
+              hideModal();
+              setEquityForSale(1);
+            }}
+            className="flex-1 flex items-center justify-center bg-blue-field/70 w-full p-4"
+          >
+            <Pressable
+              onPress={() => {}}
+              className="flex items-center w-full max-w-lg bg-blue-field p-4"
+            >
+              <Pressable
+                className="absolute right-2 top-2 "
+                onPress={() => {
+                  hideModal();
+                  setEquityForSale(1);
+                }}
+              >
+                <X color={tw.color("white")} />
+              </Pressable>
+              <Text className="text-lg font-bold">{modalText}</Text>
+              <View className="flex-row my-4 items-center">
+                <Image
+                  className="w-10 h-10"
+                  uri={entry.imageUrl ? imageUrlSmall(entry.imageUrl) : ""}
+                  fallbackUri={entry.imageUrl ? imageSrc(entry.imageUrl) : ""}
+                />
+                <Text className="ml-2">
+                  {entry.title}-{entry.artist}
+                </Text>
+              </View>
+              <Formik
+                initialValues={initialValues}
+                validationSchema={createOfferSchema}
+                validateOnMount={false}
+                onSubmit={async (values) => {
+                  setLoading(true);
+                  try {
+                    await updatePricing({
+                      variables: {
+                        id: entry.id!,
+                        equityForSale: (values.equityForSale ?? 0) / 100,
+                        price: parseInt(values.price ?? "0", 10) || 0,
+                        forSale: true,
+                        offerID: offerId,
+                      },
+                      onCompleted: onMutationCompleted,
+                    });
+                  } catch (ex) {
+                    setLoading(false);
+                    hideModal();
+                    reportError(ex);
```

Comment:
I would simply use value returned by `updatePricing` instead of using `onMutationCompleted`

Reply:
I know and you can move it all to one function like this:
```js
const { data } = await updatePricing({
  variables: {
    id: entry.id!,
    equityForSale: (values.equityForSale ?? 0) / 100,
    price: parseInt(values.price ?? "0", 10) || 0,
    forSale: true,
    offerID: offerId,
  }});
// code from onMutationCompleted
```

it will be easier to read